pf/internal/muxer/muxer.go

```diff
@@ -298,3 +299,9 @@ func copyMap[K comparable, V any](m map[K]V) map[K]V {
 	}
 	return out
 }
+
+type AliasingProviderShim struct {
+	ProviderShim
+	resources   shimUtil.AliasingResourceMap
+	datasources shimUtil.AliasingResourceMap
+}
```

Comment:
The idea here was to introduce a wrapper for `muxer.ProviderShim`, just as I did  for the base `shim.Provider`, but I took a short-cut and just modified ProviderShim directly for the POC